pkg/watcher/certificate/certificate_manager.go

```diff
@@ -0,0 +1,236 @@
+package certificate
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	apicorev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/log"
+)
+
+var (
+	ErrDomainAnnotationEmpty   = errors.New("domain annotation is empty")
+	ErrDomainAnnotationMissing = errors.New("domain annotation is missing")
+	ErrSkrCertificateNotReady  = errors.New("SKR certificate not ready")
+)
+
+//nolint:gochecknoglobals // this is const config
+var serviceSuffixes = []string{
```

Comment:
do we really need this as package level value?

Reply:
In general, I think we should avoid using global variables as much as possible, also, a string slice is not a const. I would not bother to keep this variable inside that function, if you think it's too hidden or unclear, we could also define a function to provide those values, something like:

```
func getServiceSuffix() []string {
	return []string{"svc.cluster.local", "svc"}
}
```